libpromises/mutex.c

```diff
@@ -51,11 +51,9 @@ int __ThreadLock(pthread_mutex_t *mutex,
 
     if (result != 0)
     {
-        /* Log() is blocking on mutexes itself inside malloc(), so maybe not
-         * the best idea here. */
-        Log(LOG_LEVEL_ERR,
-            "Locking failure at %s:%d function %s! (pthread_mutex_lock: %s)",
-            filename, lineno, funcname, GetErrorStrFromCode(result));
+        // printf instead of Log() to avoid problems with mutexes in Log()
+        printf("thread: %d, error: Locking failure at %s:%d function %s! (pthread_mutex_lock: %s)",
```

Comment:
I think this should be ``fprintf(stderr,...)``. And probably followed by ``fflush(); fsync()``.

Reply:
Will change. Agreed.